hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieIndexConfig.java

```diff
@@ -57,7 +64,11 @@ public class HoodieIndexConfig extends HoodieConfig {
 
   public static final ConfigProperty<String> INDEX_TYPE = ConfigProperty
       .key("hoodie.index.type")
-      .noDefaultValue()
+      .defaultValue(SIMPLE.name())
```

Comment:
There is already `HoodieIndexConfig#getDefaultIndexType` method which sets default which gets called by the builder. That method sets value based on execution engine. Why are we setting default here? I am not sure simple index (which relies on spark lean join) would work with flink. cc @danny0405

Reply:
@wzx140 i now think it should be better to leave it `noDefaultValue()` given it'll be controlled by `getDefaultIndexType()` from hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieIndexConfig.java

Reply:
@xushiyan ok. I leave it noDefaultValue() and add comment